iradix.go

```diff
@@ -357,6 +383,61 @@ func (t *Txn) delete(parent, n *Node, search []byte) (*Node, *leafNode) {
 	return nc, leaf
 }
 
+// delete does a recursive deletion
+func (t *Txn) deletePrefix(parent, n *Node, search []byte) (*Node, int) {
+	// Check for key exhaustion
+	if len(search) == 0 {
+		// Remove the leaf node
+		nc := t.writeNode(n, true)
+		subTreeSize := 0 //count this node
+		//recursively walk from all edges of the node to be deleted, tracking their mutate channels in the transaction
+		for _, e := range n.edges {
+			t.recursiveWatchNotify(e.node, func(n *Node) bool {
+				subTreeSize++
+				return false
+			})
+		}
+		if n.isLeaf() {
+			nc.leaf = nil
+			subTreeSize += 1
+		}
+		nc.edges = nil // deletes the entire subtree
+
+		return nc, subTreeSize
+	}
+
+	// Look for an edge
+	label := search[0]
+	idx, child := n.getEdge(label)
+	if child == nil || (!bytes.HasPrefix(child.prefix, search) && !bytes.HasPrefix(search, child.prefix)) {
```

Comment:
A tree with "ab", "acats", and "acato" added might be good:

```
     [root]
       | a
     [node] (prefix=a)
    b/    \c
   (ab)  [node] (prefix=cat)
        s/    \o
     (acats) (acato)
```

I think if we can run cases where we delete prefix "a", "ab", "ac", "aca", "acat", "acats", "acato" we will get the edge cases of leaves, nodes with leaves under them, nodes with nodes and leaves, and deleting a prefix that lands on a node with a longer prefix.